app/shell-config.js

```diff
@@ -8,10 +8,16 @@ var app = require("app");
 var fs = require("fs-extra");
 var path = require("path");
 var utils = require("./utils");
+var process = require("process");
+var os = require("os");
 
 var CONFIG_PATH = path.resolve(utils.convertWindowsPathToUnixPath(app.getPath("userData")), "shell-config.json");
 var config;
 
+if (!process.env["TMPDIR"] && !process.env["TMP"] && !process.env["TEMP"]) {
```

Comment:
what if one of those three is defined, and other two are not? is it enough?
shouldn't we set the remaining two to the defined one? where's the code that consumes this?